File: testdata/precision-suite-dart/clean_rawstring_blob.dart
// Clean: a long base64 `data:` URI embedded in a Dart RAW string (`r'...'`), plus
// an opaque raw-string token. Raw strings process no escapes and no
// interpolation, so lexctx classifies the whole body as a data blob — the class
// that produces the overwhelming majority of pattern false positives (a 32-char
// alphanumeric run inside an embedded base64 SVG). There is no taint flow and no
// sink here. A correct scanner emits nothing; any finding is a false positive.
class Assets {
  // A base64 data-URI blob (icon) — data, not a credential, and never a sink arg.
  static const icon =
      r'data:image/svg+xml;base64,AKIA1234567890ABCDEF1234567890ABAKIA1234567890ABCDEF1234567890AB==';

  // A raw-string regex with `$` and `\` bytes that are literal (no interpolation).
  static final idPattern = RegExp(r'^user_\$[0-9A-F]{16}$');
}

File: testdata/precision-suite-swift/clean_rawstring_blob.swift
// Clean: a base64 data-URI image embedded in a raw string literal (#"..."#) — the
// single most common secret/pattern false-positive carrier. lexctx classifies the
// whole raw string as a data blob, so pattern matches inside it are suppressed,
// and there is no taint flow. A correct scanner emits nothing; any finding here is
// a false positive.
import Foundation

// A raw string carries the blob verbatim (no escaping), spanning as one string.
let logoPNG = #"data:image/png;base64,iVBORw0KGgoAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"#

// A raw #"..."# form of a short opaque token — still data, not a live secret.
let signature = #"AKIAIOSFODNN7EXAMPLEKEYNOTREAL0000"#

func logo() -> String {
    return logoPNG
}
